source/dotnet/Library/AdaptiveCards.Templating/AdaptiveCardsTemplateVisitor.cs

```diff
@@ -92,8 +92,15 @@ public AdaptiveCardsTemplateVisitor(Func<string, object> nullSubstitutionOption,
             if (data?.Length != 0)
             {
                 // set data as root data context
-                root = JToken.Parse(data);
-                PushDataContext(data, root);
+                try
+                {
+                    root = JToken.Parse(data);
+                    PushDataContext(data, root);
+                }
+                catch (Exception e)
+                {
+                    throw new Exception("Setting root data failed with given data cotext", e);
```

Comment:
nit:

```suggestion
                    throw new Exception("Setting root data failed with given data context", e);
```

Reply:
:broken_heart: not again

---
In reply to: [424789404](https://github.com/microsoft/AdaptiveCards/pull/4011#discussion_r424789404) [](ancestors = 424789404)